File: tarai/tarai.java
public class tarai{
  public static int Tarai(int x,int y,int z){
    if(x <= y){
      return y;
    }
    else{
      return Tarai(Tarai(x-1,y,z),Tarai(y-1,z,x),Tarai(z-1,x,y));
    }
  }
  public static void main(String args []){
    long start = System.currentTimeMillis();

    Tarai(13,5,0);

    long end = System.currentTimeMillis();
    System.out.println((double)(end - start)/1000 + "s");
  }
}

File: tarai/tarai.rs
use std::time::Instant;

fn tarai(x: i32,y: i32,z: i32) -> i32 {
    if x <= y {
        return y
    }
    else{
        return tarai(tarai(x-1,y,z),tarai(y-1,z,x),tarai(z-1,x,y));
    }
}

fn main(){
    let start = Instant::now();

    tarai(13,5,0);

    let end = start.elapsed();

    println!("{}.{:3}",end.as_secs(),end.subsec_nanos() / 1_000_000);
}
